src/Tutorial.ts

```diff
@@ -0,0 +1,438 @@
+                    tooltipPopUp = tooltipPopUp + "</div>"
+                tooltipPopUp = tooltipPopUp + "</div>"
+
+            tooltipPopUp = tooltipPopUp + "</div>"
+        tooltipPopUp = tooltipPopUp + "</div>"
+
+        $('body').append(tooltipPopUp)
+    }
+
+    closeInfoPopUp = () : void =>{
+        $("#tutorialInfoPopUp").remove()
+        $(".tutorialHighlight").remove()
+    }
+
+    tutButtonNext = () : void => {
+        if(cooldown === false){
+            if(activeTutCurrentStep<activeTutStepsNo){
+                this.closeInfoPopUp()
+                activeTutCurrentStep ++
+                this.initiateTutStep('next')
+                this.startCooldown()
+            }else{
+                this.tutButtonEnd()
+            }
+        }
+    }
+
+    tutButtonPrev = () : void => {
+        if(cooldown === false){
+            if(activeTutCurrentStep>1){
+                this.closeInfoPopUp()
+                activeTutCurrentStep --
+                this.initiateTutStep('prev')
+                this.startCooldown()
+            }
+        }
+    }
+
+    tutButtonEnd = () : void => {
+        this.closeInfoPopUp()
+        $('body').off('keydown.tutEventListener');
+        $('.tutButtonListener').off('click.tutButtonListener').removeClass('tutButtonListener')
+    } 
+
+    tutPressStepListener = () : void => {
+        $('.tutButtonListener').off('click.tutButtonListener').removeClass('tutButtonListener')
+        this.tutButtonNext()
+    }
+}
+
+export class TutorialStep {
+    private title : string;
+    private text : string;
+    private type : TutorialStep.Type;
+    private waitType : Wait.Type;
+    private selector : () => void;
+    private preFunction : (eagle:Eagle) => void;
+    private backPreFunction : (eagle:Eagle) => void;
+
+    constructor(title : string, text : string, type : TutorialStep.Type,waitType: Wait.Type, selector:() => void, preFunction:(eagle:Eagle) => void, backPreFunction:(eagle:Eagle) => void){
+        this.title = title;
+        this.text = text;
+        this.type = type;
+        this.waitType = waitType
+        this.selector = selector;
+        this.preFunction = preFunction;
+        this.backPreFunction = backPreFunction;
+    }
+
+    getTitle = () : string => {
+        return this.title;
+    }
+
+    getText = () : string => {
+        return this.text;
+    }
+
+    getType = () : TutorialStep.Type => {
+        return this.type;
+    }
+
+    getWaitType = () : Wait.Type => {
+        return this.waitType;
+    }
+
+    getSelector = () : any => {
+        return this.selector;
+    }
+
+    getPreFunct = () : any => {
+        return this.preFunction;
+    }
+
+    getBackPreFunct = () : any => {
+        return this.backPreFunction;
+    }
+
+}
+
+export namespace TutorialStep {
+    export enum Type {
+        Info,
+        Press,
+        Input,
+        Condition
+    }
+}
+
+export namespace Wait {
```

Comment:
Do you need another namespace here? I think this could belong in the TutorialStep namespace.

TutorialStep.Wait.Modal etc.